zonemaster/zonemaster-backend
# Automatically create tables on damons' startup

## Purpose

This PR removes the need to explicitly run a script to create the tables during installation. Each daemon creates the tables on startup if they are not present.

## Context

This is following #834 and #841 which lay the groundwork for this PR.
This PR is based on top of #841 (which is based on top of other PRs).

## Changes

* Each daemon now run the code to create the tables and indexes at startup.

* Call `create_db()` for all engines before running the tests in `t/test01.t`.

* Remove the `share/create_db.pl` script that was introduced in #834

* Remove obsolete installation instructions introduced in #834


## How to test this PR

Try installing the Backend with the new set of instructions. You should only have to create the database (and the user, if applicable). The tables should be automatically created after starting any of the Zonemaster's daemons (`zm-rpcapi` or `zm-testagent`).
